gap/PreSheaves.gi

```diff
@@ -1366,7 +1366,9 @@ InstallMethodWithCache( PreSheavesOfFpEnrichedCategory,
                                       rho_vals[i] ) );
                 
                 mor := DirectProductFunctorial( H,
-                               mors );
+                               List( mors, Source ),
+                               mors,
+                               List( mors, Range ) );
```

Comment:
Can you get `List( mors, Source )` from `s` and `List( mors, Range )` from `r`? If yes, this would optimally use the available information.

Reply:
This can probably be done using `HomomorphismStructureOnObjects` after unpacking `s` and `r`. Maybe we can discuss this point in our next meeting. Besides, this implementation of the `HomomorphismStructure` should hopefully be soon replaced by a far superior one based on Yoneda's lemma.